conn_test.go

```diff
@@ -32,6 +33,8 @@ import (
 //	})
 //}
 
+var errHandshakeFailed = errors.New("handshake failed")
```

Comment:
Not to sure if this is good practice. Also maybe i should move this above TestNetTest().